mate-screenshot/src/mate-screenshot.c

```diff
@@ -1178,8 +1178,10 @@ load_options (void)
   /* Find various dirs */
   last_save_dir = g_settings_get_string (settings,
                                          LAST_SAVE_DIRECTORY_KEY);
-  if (!last_save_dir || !last_save_dir[0])
+
+  if (*last_save_dir == '\0')
```

Comment:
If it's guaranteed to be non-NULL, OK.  But actually if it might be you could just re-introduce the NULL check here and leave the rest as-is, as `g_free(NULL)` is perfectly valid (and just a no-op).

Reply:
https://docs.gtk.org/gio/method.Settings.get_string.html

Reply:
`g_settings_get_string` returns a newly-allocated string (`last_save_dir != NULL `will always be true), which ends with NUL (empty string check `*last_save_dir == '\0'`).